### PR TITLE
[Shop] Reduce number of queries related to product

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | no |
| BC breaks?      | yes |
| Related tickets | - |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Component/Core/Repository/ProductRepositoryInterface.php [hunk 1]
```diff
@@ -42,23 +42,19 @@ public function createShopListQueryBuilder(ChannelInterface $channel, TaxonInter
 
     /**
      * @param ChannelInterface $channel
+     * @param string$locale
```
Comment (Arminek) [inline src/Sylius/Component/Core/Repository/ProductRepositoryInterface.php]: Missing space

File: src/Sylius/Bundle/ShopBundle/Resources/views/Product/show.html.twig [hunk 2]
```diff
@@ -24,10 +24,10 @@
         <div class="ui basic segment">
             <p>{{ product.shortDescription }}</p>
         </div>
-        {% if product.isConfigurable() and product.getVariantSelectionMethod() == 'match' and not product.variants.empty() %}
-            {% include '@SyliusShop/Product/Show/_variantsPricing.html.twig' with {'pricing': sylius_product_variant_prices(product, sylius.channel)} %}
-        {% endif %}
-        {% include '@SyliusShop/Product/Show/_inventory.html.twig' %}
+        {#{% if product.isConfigurable() and product.getVariantSelectionMethod() == 'match' and not product.variants.empty() %}#}
```
Comment (pamil) [inline src/Sylius/Bundle/ShopBundle/Resources/views/Product/show.html.twig]: 🐋

File: src/Sylius/Bundle/ShopBundle/Resources/views/Product/show.html.twig [hunk 3]
```diff
@@ -24,10 +24,10 @@
...
+            {#{% include '@SyliusShop/Product/Show/_variantsPricing.html.twig' with {'pricing': sylius_product_variant_prices(product, sylius.channel)} %}#}
```
Comment (pamil) [inline src/Sylius/Bundle/ShopBundle/Resources/views/Product/show.html.twig]: 🌮

File: src/Sylius/Bundle/ShopBundle/Resources/views/Product/show.html.twig [hunk 4]
```diff
@@ -24,10 +24,10 @@
...
+        {#{% endif %}#}
```
Comment (pamil) [inline src/Sylius/Bundle/ShopBundle/Resources/views/Product/show.html.twig]: 🐐

File: src/Sylius/Bundle/ShopBundle/Resources/views/Product/show.html.twig [hunk 5]
```diff
@@ -24,10 +24,10 @@
...
+        {#{% include '@SyliusShop/Product/Show/_inventory.html.twig' %}#}
```
Comment (pamil) [inline src/Sylius/Bundle/ShopBundle/Resources/views/Product/show.html.twig]: 🌪

**Comment by pjedrzejewski** [comment]
Thanks Kamil!

**Pull request merged**